bee-message/src/payload/milestone/option/mod.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+mod pow;
+mod receipt;
+
+use derive_more::{Deref, From};
+use iterator_sorted::is_unique_sorted;
+use packable::{bounded::BoundedU8, prefix::BoxedSlicePrefix, Packable};
+pub(crate) use receipt::{MigratedFundsAmount, ReceiptFundsCount};
+
+pub use self::{
+    pow::PowMilestoneOption,
+    receipt::{MigratedFundsEntry, ReceiptMilestoneOption, TailTransactionHash},
+};
+use crate::Error;
+
+///
+#[derive(Clone, Debug, Eq, PartialEq, From, Packable)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(tag = "type", content = "data")
+)]
+#[packable(unpack_error = Error)]
+#[packable(tag_type = u8, with_error = Error::InvalidMilestoneOptionKind)]
+pub enum MilestoneOption {
```

Comment:
You probably gonna hate me for this, but the suffix `Option` is kind of a burnt term in Rust lingua. When reading other parts of the code I find myself expecting an actual Rust `Option`. But maybe that's just on my side. What about:
`MilestoneVariant`?

Reply:
It's the TIP terminology so I'd rather keep it that way